refactor(allocator): add `impl GetAddress for Address`

overlookmotel opened the pull request:
This allows passing an `Address` to methods like `StatementInjectorStore::insert_before` if you want to.

overlookmotel commented:
* **#6891** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/6891?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @overlookmotel and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

Dunqing commented:
### Merge activity

* **Oct 25, 11:19 AM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

overlookmotel committed:
refactor(allocator): add `impl GetAddress for Address` (#6891)

This allows passing an `Address` to methods like `StatementInjectorStore::insert_before` if you want to.
